config/config.go

```diff
@@ -664,6 +671,43 @@ func (c *KubernetesSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) er
 	return checkOverflow(c.XXX, "kubernetes_sd_config")
 }
 
+// EC2SDConfig is the configuration for EC2 based service discovery.
+type EC2SDConfig struct {
+	Region          string                    `yaml:"region"`
+	Access          string                    `yaml:"access,omitempty"`
+	Secret          string                    `yaml:"secret,omitempty"`
+	RefreshInterval Duration                  `yaml:"refresh_interval,omitempty"`
+	Port            int                       `yaml:"port"`
+	Tags            map[string]string         `yaml:"tags,omitempty"`
```

Comment:
I'm not sure I understand. RegexTags is a cache of Tags, but converted to regexps. Public is a user option to use the public vs private IP addresses. I'm don't understand what either of those have to do with relabeling.